src/st-trace/trace.c

```diff
@@ -98,13 +98,53 @@ static void usage(void) {
   puts("  -V, --version         Print this version");
   puts("  -vXX, --verbose=XX    Specify a specific verbosity level (0..99)");
   puts("  -v, --verbose         Specify a generally verbose logging");
-  puts("  -cXX, --clock=XX      Specify the core frequency in MHz");
-  puts("  -tXX, --trace=XX      Specify the trace frequency in Hz");
+  puts("  -cXX, --clock=XX      Specify the core frequency");
+  puts("  -tXX, --trace=XX      Specify the trace frequency");
   puts("  -n, --no-reset        Do not reset board on connection");
   puts("  -sXX, --serial=XX     Use a specific serial number");
   puts("  -f, --force           Ignore most initialization errors");
 }
 
+static bool parse_frequency(char* text, uint32_t* result, double default_scale)
+{
+  if (text == NULL) {
+    ELOG("Invalid frequency.\n");
+    return false;
+  }
+
+  char* suffix = text;
+  double value = strtod(text, &suffix);
+
+  if (value == 0.0) {
+    ELOG("Invalid frequency.\n");
+    return false;
+  }
+
+  for (int i = 0; suffix[i] != 0; i++)
+    suffix[i] = tolower(suffix[i]);
+
+  double scale = default_scale;
+  if (strcmp(suffix, "hz") == 0)
```

Comment:
One should use the correct upper/lower case orthography here: Hz, kHz, MHz and GHz

Reply:
No, not necessarily, but the implementation should interpret it correctly and ensure that e.g. "khz" and "kHz" is recognised as an input, but always displayed as "kHz" in any output wherever it may occur.